src/test/java/com/chromediopside/controller/MatchControllerTest.java

```diff
@@ -0,0 +1,96 @@
+package com.chromediopside.controller;
+
+import static org.junit.Assert.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.webAppContextSetup;
+
+import com.chromediopside.GitinderApplication;
+import com.chromediopside.mockbuilder.MockUserBuilder;
+import com.chromediopside.model.Swiping;
+import com.chromediopside.repository.SwipeRepository;
+import com.chromediopside.repository.UserRepository;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = GitinderApplication.class)
+@AutoConfigureMockMvc
+@WebAppConfiguration
+@EnableWebMvc
+public class MatchControllerTest {
+
+  private MockMvc mockMvc;
+
+  @Autowired
+  private WebApplicationContext webApplicationContext;
+
+  @MockBean
+  private UserRepository userRepository;
+  @MockBean
+  private SwipeRepository swipeRepository;
+
+  @Autowired
+  private MockUserBuilder mockUserBuilder;
+
+  @Before
+  public void setup() throws Exception {
+    this.mockMvc = webAppContextSetup(webApplicationContext).build();
+  }
+
+  @Test
+  public void getMatches() throws Exception {
+
+    Mockito.when(userRepository.findByAppToken("321")).thenReturn(
+            mockUserBuilder.setAccessToken("123")
+                    .setAppToken("321")
+                    .setUserName("nori")
+                    .build());
+
+    Timestamp timestamp = new Timestamp(1498563273079l);
+    Swiping s1 = new Swiping("nori", "dani", timestamp);
+    Swiping s2 = new Swiping("nori", "dori", timestamp);
+    Swiping s3 = new Swiping("peti", "nori", timestamp);
```

Comment:
this would be much better if these objects were created in a separate class and injected here.